File: backend/src/main/java/com/go_exchange_easier/backend/dto/details/UniversityReviewDetails.java
package com.go_exchange_easier.backend.dto.details;

import com.go_exchange_easier.backend.dto.summary.UniversitySummary;
import com.go_exchange_easier.backend.dto.summary.UserWithAvatarSummary;
import java.time.Instant;
import java.util.List;

public record UniversityReviewDetails(

        Integer id,
        UserWithAvatarSummary author,
        UniversitySummary university,
        Short starRating,
        String textContent,
        Instant createdAt,
        List<ReactionDetails> reactions

) { }

File: frontend/src/dtos/details/UniversityReviewDetails.ts
import type { UniversitySummary } from "../summary/UniversitySummary";
import type { UserWithAvatarSummary } from "../summary/UserWithAvatarSummary";
import type { ReactionDetails } from "./ReactionDetails";

export type UniversityReviewDetails = {
  id: number;
  author: UserWithAvatarSummary;
  university: UniversitySummary;
  starRating: number;
  textContent: string;
  createdAt: string;
  reactions: ReactionDetails[];
};
